gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/CircularDependencyChecker.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.common.impl;
+
+import gobblin.config.store.api.ConfigStore;
+import gobblin.config.store.api.VersionDoesNotExistException;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.hadoop.fs.Path;
+
+
+public class CircularDependencyChecker {
```

Comment:
I agree. There should be two circular dependency checkers. One should be used during import resolution. It's a matter of simply keeping state of which URL have already been imported. When we see a new import we check if it is in the "seen" set. If yes, there is a circular dependency, otherwise add it to the set. The is obviously P1.

The second implementation is  a store validation that can be run when we build a new config artifact for deployment. In this case, we need to check for any cycles. This is P2.

Further, I don't know why all the methods are static. There is obviously some state, say the ConfigStore and versio that is carried around. This should be put in object state.